pkg/kubelet/kubelet.go

```diff
@@ -1789,6 +1789,9 @@ func (kl *Kubelet) syncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	// Fetch the pull secrets for the pod
 	pullSecrets := kl.getPullSecretsForPod(pod)
 
+	// Ensure the pod is being probed
+	kl.probeManager.AddPod(pod)
```

Comment:
This method will log errors (but not fail) when it is called multiple times.  However, it will *not* check for readiness probes if it is called a second time on startupprobes. Something you want to address?

Reply:
podWorkers knows when to call syncPod. This should be ok.